distributed/core.py

```diff
@@ -348,29 +490,30 @@ def start_periodic_callbacks(self):
         """Start Periodic Callbacks consistently
 
         This starts all PeriodicCallbacks stored in self.periodic_callbacks if
-        they are not yet running.  It does this safely on the IOLoop.
+        they are not yet running. It does this safely by checking that it is using the
+        correct event loop.
         """
-        self._last_tick = time()
-
-        def start_pcs():
-            for pc in self.periodic_callbacks.values():
-                if not pc.is_running():
-                    pc.start()
+        if self.io_loop.asyncio_loop is not asyncio.get_running_loop():
+            raise RuntimeError(f"{self!r} is bound to a different event loop")
 
-        self.io_loop.add_callback(start_pcs)
+        self._last_tick = time()
+        for pc in self.periodic_callbacks.values():
+            if not pc.is_running():
+                logger.info(f"Starting periodic callback {pc!r}")
```

Comment:
I think this might be causing the failures in test_worker_cli_nprocs_renamed_to_nworkers see also https://github.com/dask/distributed/pulls/

```suggestion
```